src/tests/ftest/util/fio_test_base.py

```diff
@@ -21,6 +21,7 @@ def __init__(self, *args, **kwargs):
         self.fio_cmd = None
         self.processes = None
         self.manager = None
+        self.dfuse_bind_cores = None
```

Comment:
Since `FioBase` inherits `DfuseTestBase`, it might be better to move this line to `DfuseTestBase.__init__` so other tests can take advantage. `DfuseTestBase.start_dfuse` could default `bind_cores` to `self.dfuse_bind_cores` so `FioBase` doesn't have to know anything about it.

Reply:
In that case we probably don't need to take it as a parameter to the run() method, as it can read from self?

Ideally there would be a way of loading this setting from the yaml file but how to do that escapes me.  Maybe something with BasicParameter?

Reply:
`BasicParameter` is a command parameter. At the level of `DfuseCommand`, this wouldn't quite work because of the way we run the commands. But... you can do something similar to `ior_timeout` here: https://github.com/daos-stack/daos/blob/873844e5c351af24655c50f396a755e558ea8044/src/tests/ftest/util/ior_test_base.py#L55
In `DfuseTestBase.setup()`: `self.dfuse_bind_cores= self.params.get("bind_cores", self.dfuse.namespace, None)`
And `DfuseTestBase.start_dfuse()`: `self.dfuse.run(self.dfuse_bind_cores)`

Reply:
Done, now back into the debugging phase.

Would it be better to do this in DfuseTestBase or DfuseCommand do you think?  You put DfuseTestBase and that's what I went with but all the other options are in DfuseCommand and DfuseTestBase doesn't do anything with it rather than pass it on.

One nice to have would be that the yaml files could define a default under the dfuse section, but then also have a mux in the test section to be able to try different values, that would probably need DfuseCommand and DfuseTestBase code, both reading from their own namespace?

Reply:
It's a little messy, and I'm probably not explaining it well, but... DfuseCommand can really only get yaml values for params that are actual `dfuse` params. For example, if you put `self.cores = FormattedParameter('--cores {}')` in DfuseCommand, the command would look like `dfuse --cores <cores>`. So anything outside the actual dfuse command needs to be passed to the run method or job manger. You could maybe have a "normal" variable in DfuseCommand to hold cores, though I'm not sure we do that anywhere else.

> DfuseCommand and DfuseTestBase code, both reading from their own namespace

Right now, I think both use the `dfuse` namespace, so it would be tricky. Generally, tests will define a mux under some other namespace, and read + pass/set the values manually in the test code. There are ways to improve this, but the command code is a messy IMO.